src/App.tsx

```diff
@@ -15,49 +16,119 @@ export const goodsFromServer = [
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case (SortType.ALPABET): {
+      visibleGoods.sort((first, second) => first.localeCompare(second));
+      break;
+    }
+
+    case (SortType.LENGTH): {
+      visibleGoods.sort((first, second) => first.length - second.length);
+      break;
+    }
+
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    return visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const resetButtonRules = sortType !== SortType.NONE || isReversed;
+  const alphabetButtonRules = sortType !== SortType.ALPABET;
+  const lengthButtonRules = sortType !== SortType.LENGTH;
+
+  const preparedGoods = getReorderedGoods(
+    goodsFromServer,
+    { sortType, isReversed },
+
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames(
+            'button is-info',
+            { 'button is-info is-light': alphabetButtonRules },
+          )}
+          onClick={() => setSortType(SortType.ALPABET)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={classNames(
+            'button is-success',
+            { 'button is-success is-light': lengthButtonRules },
+          )}
+          onClick={() => setSortType(SortType.LENGTH)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={classNames(
+            'button is-warning',
+            { 'button is-warning is-light': !isReversed },
+          )}
+          onClick={() => setIsReversed(!isReversed)}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {resetButtonRules && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setIsReversed(false);
+              setSortType(SortType.NONE);
+            }}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {preparedGoods.map(good => (
+          <li
+            key={good}
+            data-cy="Good"
+          >
```

Comment:
```suggestion
          <li key={good} data-cy="Good">
```
if you have less than 3 attributes you can leave them on the same line